pkg/pricer/pricer.go

```diff
@@ -198,17 +239,67 @@ func (s *Pricer) NotifyPriceTable(peer swarm.Address, priceTable []uint64) error
 
 	pricingPeer.lock.Lock()
 	defer pricingPeer.lock.Unlock()
-	s.logger.Debugf("Storing pricetable %v for peer %v", priceTable, peer)
-	err = s.store.Put(peerPriceTableKey(peer), priceTable)
+
+	return s.storePriceTable(peer, priceTable)
+}
+
+func (s *Pricer) NotifyPeerPrice(peer swarm.Address, price uint64, index uint8) error {
+
+	if price == 0 {
+		return nil
+	}
+
+	pricingPeer, err := s.getPricingPeer(peer)
 	if err != nil {
-		return fmt.Errorf("failed to persist pricetable for peer %v: %w", peer, err)
+		return err
 	}
 
-	return nil
+	pricingPeer.lock.Lock()
+	defer pricingPeer.lock.Unlock()
+
+	priceTable := s.PeerPriceTable(peer)
+	currentIndexDepth := uint8(len(priceTable)) - 1
+
+	if index <= currentIndexDepth {
+		// Simple case, already have index depth, single value change
+		priceTable[index] = price
+		s.logger.Debugf("Storing updated pricetable %v for peer %v", priceTable, peer)
+		return s.storePriceTable(peer, priceTable)
+	}
+
+	// Complicated case, index is larger than depth of already known table
+	newPriceTable := make([]uint64, index+1)
+
+	// Copy previous content
+	copied := copy(newPriceTable, priceTable)
+
+	// sanity check copy was successful
+	if uint8(copied) != currentIndexDepth+1 {
```

Comment:
what is this for? remove if only for debugging

Reply:
Not sure we want to sanity check this, removed this for now